### DISCUSSION TITLE
Quarkus rest-client reactive - Is it possible to filter a response after ReaderInterceptor ?

### DISCUSSION THREAD

**Discussion by Sirz3chs** [quarkusio/quarkus#32062] [Q&A]
Hi, to provide some context, I am currently developing a custom extension that generates audit logs for REST clients. The extension involves registering the following components:

- A `ResteasyReactiveClientRequestFilter` to initialize the log with request metadata, invoked method, and request body.
- A `ResteasyReactiveClientResponseFilter` to define the response metadata.
- A `ReaderInterceptor` to obtain the response body and save the log.
- A `ResteasyReactiveResponseExceptionMapper` that calls `response.readEntity()` to invoke the `ReaderInterceptor` and obtain the exception to continue the chain.

This flow works well, where the request succeeds by going through `request filter` -> `response filter` -> `reader interceptor`. However, when the request returns `void`, such as with a `204 - Created`, the `ReaderInterceptor` is not invoked, and my audit log is not saved.

I would like to know if there is a filter or another component that applies after the `ReaderInterceptor` to save my audit log, even in cases where the `ReaderInterceptor` is not invoked.

[a comment by an automated account is not shown]

**Comment by geoand** [comment]
Hi,

There is no such component unfortunately. The lifecycle is essentially determined by the JAX-RS (Jakarta REST) spec.

**Reply by Sirz3chs** [reply]
Thanks for the answer, I managed to find a solution to my problem thanks to the `hasEntity()` method on the `ClientResponseContext`.
